test/performance/deployment-probe/main.go

```diff
@@ -186,102 +190,80 @@ func main() {
 	defer paWI.Stop()
 	paSeen := sets.String{}
 
-	// Interval for creating services.
-	tick := time.NewTicker(5 * time.Second)
-
-LOOP:
-	for {
-		select {
-		case <-ctx.Done():
-			// If we timeout or the pod gets shutdown via SIGTERM then start to
-			// clean thing up.
-			break LOOP
-
-		case ts := <-tick.C:
-			svc, err := sc.ServingV1beta1().Services(tmpl.Namespace).Create(tmpl)
-			if err != nil {
-				q.AddError(mako.XTime(ts), err.Error())
-			}
-			log.Printf("Created: %s", svc.Name)
-
-		case event, ok := <-serviceWI.ResultChan():
-			if !ok {
-				fatalf("Unexpected end of watch")
-			}
-			if event.Type != watch.Modified {
-				// Skip events other than modifications
-				break
-			}
-			svc := event.Object.(*v1beta1.Service)
-			handle(q, svc, svc.Status.Status, &serviceSeen, "dl")
-
-		case event, ok := <-configurationWI.ResultChan():
-			if !ok {
-				fatalf("Unexpected end of watch")
-			}
-			if event.Type != watch.Modified {
-				// Skip events other than modifications
-				break
-			}
-			cfg := event.Object.(*v1beta1.Configuration)
-			handle(q, cfg, cfg.Status.Status, &configurationSeen, "cl")
-
-		case event, ok := <-routeWI.ResultChan():
-			if !ok {
-				fatalf("Unexpected end of watch")
-			}
-			if event.Type != watch.Modified {
-				// Skip events other than modifications
-				break
-			}
-			rt := event.Object.(*v1beta1.Route)
-			handle(q, rt, rt.Status.Status, &routeSeen, "rl")
-
-		case event, ok := <-revisionWI.ResultChan():
-			if !ok {
-				fatalf("Unexpected end of watch")
+	tick := time.NewTicker(frequency)
+	func() {
+		for {
+			select {
+			case <-ctx.Done():
+				// If we timeout or the pod gets shutdown via SIGTERM then start to
+				// clean thing up.
+				return
+
+			case ts := <-tick.C:
+				svc, err := sc.ServingV1beta1().Services(tmpl.Namespace).Create(tmpl)
+				if err != nil {
+					q.AddError(mako.XTime(ts), err.Error())
+				}
+				log.Printf("Created: %s", svc.Name)
+
+			case event := <-serviceWI.ResultChan():
+				if event.Type != watch.Modified {
```

Comment:
It's easier to go the other way around and just do
`if modified {}`.

Reply:
Seems more go-stylish to avoid the indentation on the happy path, either way it doesn't end up being that big a difference.  I think if we could shrink things to a single guarded call I'd be more inclined to go that way 

LMK if you really care 🤷‍♂